tests/tuning/hyperparams/test_boolean.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+from unittest import TestCase
+
+from btb.tuning.hyperparams.boolean import BooleanHyperParam
+
+
+class TestBooleanHyperParam(TestCase):
+    """Unit test for the class ``BooleanHyperParam``."""
+
+    def setUp(self):
+        self.instance = BooleanHyperParam()
+
+    def test_trasnform_list(self):
+        """Test that the method ``transform`` performs a normalization over a ``list`` of boolean
+        values and converts them in to a search space of [0, 1]^k.
+        """
+
+        # setup
+        values_1 = [True, False, True]
+        values_2 = [False, False, True]
+
+        # run
+        result_1 = self.instance.transform(values_1)
+        result_2 = self.instance.transform(values_2)
+
+        # assert
+        assert all(isinstance(res[0], int) for res in result_1.tolist())
+        assert all(isinstance(res[0], int) for res in result_2.tolist())
+
+    def test_transform_scalar(self):
+        """Test that the method ``transform`` performs a normalization over a sinlge boolean
+        value and converts them in to a search space of [0, 1]^k.
+        """
+
+        # setup
+        values_1 = False
+        values_2 = True
+
+        # run
+        result_1 = self.instance.transform(values_1)
+        result_2 = self.instance.transform(values_2)
+
+        # assert
+        assert all(isinstance(res[0], int) for res in result_1.tolist())
```

Comment:
the fact that this assertion is so elaborate suggests to me this unit test could be improved in design